src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -95,23 +95,21 @@ async def generate_python_from_protobuf(
         target_stripped_sources_request,
     )
 
-    # To run the MyPy Protobuf plugin, we first install it with Pex, then extract the wheels and
-    # point Protoc to the extracted wheels with its `--plugin` argument.
-    extracted_mypy_wheels = None
+    protoc_gen_mypy_script = "protoc-gen-mypy"
+    mypy_pex = None
     if python_protobuf_subsystem.mypy_plugin:
         mypy_pex = await Get(
-            Pex,
-            PexRequest(
-                output_filename="mypy_protobuf.pex",
-                internal_only=True,
-                requirements=PexRequirements([python_protobuf_subsystem.mypy_plugin_version]),
-                # This is solely to ensure that we use an appropriate interpreter when resolving
-                # the distribution. We don't actually run the distribution directly with Python,
-                # as we extract out its binary.
```

Comment:
Some portion of this comment is probably worth preserving. Why did we choose 3.5+?

Reply:
We was you. Why did you choose it? I assumed it was because MyPy requires 3.5+ ("... ensure that we use an appropriate interpreter ..."). That comment really explained nothing - how is >=3.5 appropriate?

Reply:
Hrm - and it looks false to me even: https://github.com/dropbox/mypy-protobuf/tree/v1.23
The code claims it can work with 2.7+.

Reply:
So I actually have no clue why we restrict to 3.5+. A valid explanation would be great. I'm happy to add it if you've got one. Otherwise I'll try forcing 2.7.* to prove / see if that works and then ... I'm not sure. Perhaps this deserves an issue and a a follow up. Its not clear at all to me this is / was correct to hardcode. The current release has a python_requires of >=3.6 for example which this breaks.

Reply:
Yeah, so this works fine:
```diff
$ git diff
diff --git a/src/python/pants/backend/codegen/protobuf/python/rules.py b/src/python/pants/backend/codegen/protobuf/python/rules.py
index 4bf6e3fcd..523079a5d 100644
--- a/src/python/pants/backend/codegen/protobuf/python/rules.py
+++ b/src/python/pants/backend/codegen/protobuf/python/rules.py
@@ -106,7 +106,7 @@ async def generate_python_from_protobuf(
                     output_filename="mypy_protobuf.pex",
                     internal_only=True,
                     requirements=PexRequirements([python_protobuf_subsystem.mypy_plugin_version]),
-                    interpreter_constraints=PexInterpreterConstraints(["CPython>=3.5"]),
+                    interpreter_constraints=PexInterpreterConstraints(["CPython==2.7.*"]),
                 ),
             ),
         )
```

I'll file an issue and add a TODO pointing to that.

Reply:
Huh, you're right. I think it was based on MyPy requiring 3.5+ but indeed that is super bogus here. We shouldn't care about MyPy, only what this dist itself requires.

Reply:
#11565

Reply:
Added a comment that explains the current guess approach and the fix with issue link.